### DISCUSSION TITLE
How do you implement pager when you want to specify pages in config?

### DISCUSSION THREAD

**Discussion by zachbogart** [observablehq/framework#1100] [Q&A] [answered]
I'm having trouble getting the `pager` feature to work when I specify the pages in the config. 

For example, I am specifying the [config](https://github.com/cookwcode/restaurant/blob/main/observablehq.config.ts) to show the demo pages within a folder. I see the [first footer pager](https://cookwcode.observablehq.cloud/restaurant/), which links to the next page, all subsequent pages don't have pagers. This only seems to happen if I specify the pages in the config (if I remove the pages field entirely, all the pager footer links show up across the board, but my contents now are no longer nested in the sidebar).

Any ideas about why the pager links aren't showing up for all pages if I specify the page listings in the config?

Thanks.

**Comment by Fil** [comment]
Framework's [documentation](https://observablehq.com/framework/) uses the **pages** option and has a pager. The code is here:
https://github.com/observablehq/framework/blob/main/observablehq.config.ts#L4
It should be similar for your site. ~~If you think you found a bug please share a reproducible case.~~

**Reply by mbostock** [reply]
It seems reproducible in the linked repo. We should investigate.

**Reply by Fil** [reply]
Oops sorry I looked at the links too quickly.

**Reply by Fil** [reply, marked as answer]
The problem comes from the fact that the paths are specified in **pages** without a leading /.

**Reply by Fil** [reply]
We should update https://observablehq.com/framework/config#pages to show actual pages instead of the typescript "string"

**Reply by mbostock** [reply]
@Fil So we’re missing some additional path normalization when parsing the config, it seems?

**Reply by Fil** [reply]
Yes (tracking at #1110); I also want more explicit documentation in #1109.
